Doc/library/xml.etree.elementtree.rst

```diff
@@ -638,6 +638,83 @@ Functions
+
+Example
+^^^^^^^
+
+Here's an example that demonstrates use of the XInclude module. To include a XML document in the current document, use the ``{http://www.w3.org/2001/XInclude}include`` element and set the **parse** attribute to ``“xml”``, and use the **href** attribute to specify the document to include.::
+
+   <document xmlns:xi="http://www.w3.org/2001/XInclude">
+   <xi:include href="source.xml" parse="xml" />
+   </document>
+
+By default, the **href** attribute is treated as a file name. You can use custom loaders to override this behaviour. Also note that the standard helper does not support XPointer syntax.
+
+To process this file, load it as usual, and pass the root element to the **ElementTree** module: ::
+
+   from elementtree import ElementTree, ElementInclude
+
+   tree = ElementTree.parse("document.xml")
+   root = tree.getroot()
+
+   ElementInclude.include(root)
+
+The ElementInclude module replaces the ``{http://www.w3.org/2001/XInclude}include`` element with the root element from the **source.xml** document. The result might look something like this: ::
+
+   <document xmlns:xi="http://www.w3.org/2001/XInclude">
+   <para>This is a paragraph.</para>
+   </document>
+
+If the **parse** attribute is omitted, it defaults to “xml”. The href attribute is required.
+
+To include a text document, use the ``{http://www.w3.org/2001/XInclude}include`` element, and set the **parse** attribute to “text”: ::
+
+   <document xmlns:xi="http://www.w3.org/2001/XInclude">
+   Copyright (c) <xi:include href="year.txt" parse="text" />.
+   </document>
+
+The result might look something like: ::
+
+   <document xmlns:xi="http://www.w3.org/2001/XInclude">
+   Copyright (c) 2003.
+   </document>
+
+Reference
+---------
+
+.. _elementtree-functions:
+
+Functions
+^^^^^^^^^
+
+.. function:: xml.etree.ElementInclude.default_loader( href, parse, encoding = None)
+
+   This loader reads an included resource from disk.  *href* is a resource
+   reference.  *parse* is for parse mode either "xml" or "text".  *encoding* 
+   is an optional text encoding.  If not given, none is used.  Returns the 
+   expanded resource.  If the parse mode is ``"xml"``, this is an ElementTree 
+   instance.  If the parse mode is "text", this is an Unicode string.  If the 
+   loader fails, it can return None or raise an **IOError** exception.
+
+
+.. function:: xml.etree.ElementInclude.include( elem, loader = None)
+
+   This expand XInclude directives.  *elem* is the root element.  *loader* is 
+   an optional resource loader.  If omitted, it defaults to default_loader.  
+   If given, it should be a callable that implements the same interface as 
+   default_loader.  Returns the expanded resource.  If the parse mode is 
+   ``"xml"``, this is an ElementTree instance.  If the parse mode is "text", 
+   this is an Unicode string.  If the loader fails, it can return None or 
+   raise an **IOError** exception.
```

Comment:
I wouldn't specify the exact exception here, although suggesting it as a common one seems ok.